discussment-ui-wicket/src/main/java/org/danekja/discussment/ui/wicket/list/discussion/DiscussionListPanel.java

```diff
@@ -80,32 +90,46 @@ protected void populateItem(final ListItem<Discussion> listItem) {
                     listItem.add(new Label("numberOfPosts"));
                 }
 
-                listItem.add(new Label("lastUsername", new LoadableDetachableModel<String>() {
-                    protected String load() {
-                        try {
-                            return discussionService.getLastPostAuthor(listItem.getModelObject()).getDisplayName();
-                        } catch (DiscussionUserNotFoundException e) {
-                            return getString("error.userNotFound");
-                        } catch (AccessDeniedException e){
-                            return getString("error.accessDenied");
-                        } catch (NullPointerException e){
-                            return "";
-                        }
-                    }
-                }));
-                try {
-                    listItem.add(new Label("lastCreated", postService.getLastPost(listItem.getModelObject()).getCreated()));
-                } catch (NullPointerException e){
-                    listItem.add(new Label("lastCreated"));
-                } catch (AccessDeniedException e){
+                Post lastPost = getLastPost(listItem.getModelObject());
+                if (lastPost != null) {
+                    listItem.add(new Label("lastCreated", lastPost.getCreated()));
+                } else {
                     listItem.add(new Label("lastCreated"));
                 }
 
+                IDiscussionUser lastPostAuthor = getPostAuthor(lastPost);
+                if (lastPostAuthor != null) {
+                    listItem.add(new Label("lastUsername", lastPostAuthor.getDisplayName()));
+                } else {
+                    listItem.add(new Label("lastUsername"));
+                }
+
                 listItem.add(createRemoveDiscussionLink(listItem.getModel()));
             }
         });
+
+        add(new PasswordForm("passwordForm", discussionModel, new Model<>(new Discussion()), userService, accessControlService, permissionService, discussionService));
+    }
+
+    private Post getLastPost(Discussion discussion) {
```

Comment:
I believe this method should be replaced by a Model which will be passed to Label (and either show something or not)

Reply:
True.